backend/app/services/scenario_preparation_service.py

```diff
@@ -247,12 +269,23 @@ def generate_scenario_preparation(
             other_party=request.other_party,
         )
 
+        vector_db_prompt_extension = query_vector_db_and_prompt(
```

Comment:
I think that vector_db_prompt_extension may be a bit confusing. If I understand it correctly, it is just the fixed objective for every session, right? I'd prefer a name such as _session_objective_ or _fixed_session_objective_ instead. If you agree, please refactor it for all methods